modules/icinga/manifests/service_dependency.pp

```diff
@@ -0,0 +1,89 @@
+# == Define: icinga::service_dependency
+#
+# This creates a service dependency for the host service that other host
+# services are dependent on.
+#
+# Most of these params relate to Nagios service configuration directives:
+#
+# https://assets.nagios.com/downloads/nagioscore/docs/nagioscore/3/en/objectdefinitions.html#servicedependency
+#
+# === Parameters:
+#
+# [*host_name*]
+#   The title of a `Icinga::Host` resource. Usually `$::fqdn`.
+#
+#   This is a mandatory argument because the type is typically used as an
+#   exported resource. In which case the variable must be eagerly evaluated
+#   when passed by the exporting node, rather than lazily evaluated inside
+#   the define by the collecting node.
+#
+# [*ensure*]
+#   Can be used to remove an existing check.
+#   Default: present
+#
+# [*service_description*]
+#   Description of the service that is dependent services rely upon.
+#
+# [*check_command*]
+#   Command used to check whether the service is healthy.
+#
+# [*dependent_host_name*]
+#   This host is dependent on `host_name`.
+#
+# [*dependent_service_decription*]
+#   Description of the service where functionality is changed depending on the
+#   status of `service_description`.
+#
+# [*execution_failure_criteria*]
+#   A comma-seperated set of criteria that determine what state the service is
+#   in for how actively the dependent service is checked.
+#   Default: 'n': none (services are always checked)
+#
+# [*notifcation_failure_criteria*]
+#   A comma-seperated set of criteria that determines when notifcations should
+#   suppressed for the dependent service.
+#   Default: 'w,c': warning state, critical state
+#
+define icinga::service_dependency (
+  $host_name,
+  $ensure                         = 'present',
+  $service_description            = undef,
+  $dependent_host_name            = undef,
+  $dependent_service_description  = undef,
+  $execution_failure_criteria     = 'n',
+  $notification_failure_criteria  = 'w,c',
+) {
+
+  validate_re($dependent_service_description, '^(\w|\s|\-|/|\[|\]|:|\.)*$', "Icinga check \"${dependent_service_description}\" contains invalid characters")
+
+  $check_filename = "/etc/icinga/conf.d/icinga_host_${host_name}/${title}.cfg"
+
+  if $ensure == 'present' {
+
+    if $service_description == undef {
+      fail("Must provide a \$service_description to icinga::Service_dependency[${title}]")
+    }
+    if $dependent_service_description == undef {
+      fail("Must provide a \$dependent_service_description to icinga::Service_dependency[${title}]")
+    }
+
+    if $dependent_host_name == undef {
+      fail("Must provide a \$dependent_host_name to icinga::Service_dependency[${title}]")
+    }
+
+    $check_content = template('icinga/service_dependency.erb')
+
+    Icinga::Host[$host_name] -> Icinga::Service_dependency[$title]
+
+  } else {
+    $check_content = ''
+  }
+
+  file { $check_filename:
```

Comment:
Should we write an essentially empty config file out or should this resource be inside the if condition?